### DISCUSSION TITLE
Upgrade process for Linux

### DISCUSSION THREAD

**Discussion by canuckcam** [chuot/rdio-scanner#90] [Q&A]
As I stumble around Linux, would this be the proper way to upgrade the linux version of rdio-scanner? I guess my main question is that I can unzip the upgraded executable on top of the existing without stopping the service first (minimal downtime?) and whether that's ok? The config is saved in the .ini so everything is retained.

1. SSH into the server.
2. Go into the rdio-scanner directory.
`$ cd ~/rdio-scanner`
3. On a desktop browser, go to https://github.com/chuot/rdio-scanner/releases and right-click the link to the latest release to copy the URL.
4. Download radio-scanner .zip file.
`$ wget https://github.com/chuot/rdio-scanner/releases/download/v6.1.3/rdio-scanner-linux-amd64-v6.1.3.zip`
5. Unzip the downloaded file to the current directory, overwriting the old executable.
`$ unzip rdio-scanner-linux-amd64-v6.1.3.zip
Archive:  rdio-scanner-linux-amd64-v6.1.3.zip replace rdio-scanner? [y]es, [n]o, [A]ll, [N]one, [r]ename: A 
`
6. Restart rdio-scanner service.
`$ ./rdio-scanner -service restart`

**Comment by chuot** [comment]
Unfortunately, downtime is innevitable. First, you need to take a backup of your current instance prior to attempting an upgrage. Then follow the instructions in [docs/update-from-v5.md](https://github.com/chuot/rdio-scanner/blob/master/docs/update-from-v5.md).
